docker_custodian/docker_gc.py

```diff
@@ -46,16 +48,21 @@ def is_container_old(container, min_date):
     if state.get('Ghost'):
         return True
 
+    created_date = dateutil.parser.parse(container['Created'])
+
+    # Don't delete recently created containers
+    if max_date and created_date > max_date:
+        return False
+
     # Container was created, but never used
-    if (state.get('FinishedAt') == YEAR_ZERO and
-            dateutil.parser.parse(container['Created']) < min_date):
+    if state.get('FinishedAt') == YEAR_ZERO and created_date < min_date:
```

Comment:
It is not straight-forward to reproduce since the sequence of events are interleaved between the dcds client and another docker client creating/starting containers with the state changing continuously. I'll see if I can dig up some more info.

```
# container created by docker client
2015-12-13T10:05:01-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:01-08:00" level=debug msg="Calling POST /containers/create" 
2015-12-13T10:05:01-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:01-08:00" level=info msg="POST /v1.18/containers/create?name=yelpmainsandboxsandboxymsspatel1271_memcache_1" 
2015-12-13T10:05:01-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:01-08:00" level=info msg="+job create(yelpmainsandboxsandboxymsspatel1271_memcache_1)" 

# container deleted by dcds
2015-12-13T10:05:01-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:01-08:00" level=info msg="GET /v1.15/containers/c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e/json" 
2015-12-13T10:05:01-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:01-08:00" level=info msg="+job container_inspect(c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e)" 
2015-12-13T10:05:01-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:01-08:00" level=info msg="-job container_inspect(c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e) = OK (0)" 
2015-12-13T10:05:01-08:00 <user.notice> <redacted> cron_purge-old-docker-artifacts: Removing container c4ba7c934840eb9a yelpmainsandboxsandboxymsspatel1271_memcache_1 0001-01-01T00:00:00Z
2015-12-13T10:05:02-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:02-08:00" level=info msg="+job log(destroy, c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e, redacted/memcache-testing:latest)" 
2015-12-13T10:05:02-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:02-08:00" level=info msg="-job log(destroy, c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e, redacted/memcache-testing:latest) = OK (0)" 
2015-12-13T10:05:02-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:02-08:00" level=info msg="-job rm(c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e) = OK (0)" 

# container expected to exist
2015-12-13T10:05:03-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:03-08:00" level=info msg="POST /v1.18/containers/c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e/start" 
2015-12-13T10:05:03-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:03-08:00" level=info msg="+job start(c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e)" 
2015-12-13T10:05:03-08:00 <daemon.info> <redacted> dockerd: no such id: c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e
2015-12-13T10:05:03-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:03-08:00" level=info msg="-job start(c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e) = ERR (1)" 
2015-12-13T10:05:03-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:03-08:00" level=error msg="Handler for POST /containers/{name:.*}/start returned error: no such id: c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e" 
2015-12-13T10:05:03-08:00 <daemon.info> <redacted> dockerd: time="2015-12-13T10:05:03-08:00" level=error msg="HTTP Error: statusCode=404 no such id: c4ba7c934840eb9a9b7a7ff699f69acd86568dbc5006f278cae52be8c688078e" 
```